micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java

```diff
@@ -199,6 +209,13 @@ private void stopTimed(String metricName, Timer.Sample sample,
             final Timer timer = Timer.builder(metricName)
                     .description(description)
                     .tags(tags)
+                    .tags(
+                        tagsBasedOnMethodInvocationContext
+                            .stream()
+                            .flatMap(b -> Stream.of(b.buildTags(context)))
```

Comment:
add a null check with a good error can case `buildTags` returns `null`

Reply:
would this suggestion still be needed if `buildTags` is annotated with `@NonNull`?

Reply:
yes because people do stupid things